src/binfhe/include/binfhecontext.h

```diff
@@ -411,7 +411,7 @@ class BinFHEContext : public Serializable {
     std::map<uint32_t, RingGSWEvalKey> m_BTKey_map;
 
     // Whether to optimize time for sign eval
-    bool m_timeOptimization;
+    bool m_timeOptimization = false;
```

Comment:
This data member should be initialized in constructors.

Reply:
Sorry, I may be wrong as  C++ may allow initialization of non-static or not-const data members this way. I need to double check this

Reply:
Yes, I am wrong. It is allowed

Reply:
If you're happy with it as is would you mind merging it? It sounds like you are